src/utils/components/ModalProvider/ModalProvider.tsx

```diff
@@ -47,7 +47,7 @@ export const useModalValue = (): ModalContextType => {
 
   const createModal = (newModal: ModalComponent) => {
     setIsOpen(true);
-    setModal(() => newModal);
+    setModal(() => newModal); // TODO: possible bug? should be 'setModal(newModal)'
```

Comment:
what is this todo?

Reply:
I was wondering why we're not setting the modal directly like
```ts
setModal(newModal);
```
but doing that caused some bugs in modals so we reverted this change.

Technically, `setModal` is React `useState` setter function and `() => newModal` should be equivalent to `(prevModal) => newModal` where `prevModal` is unused.